plugins/tableselection/plugin.js

```diff
@@ -311,12 +311,20 @@
 
 			// Covers a case when:
 			// 1. User releases mouse button outside the table.
-			// 2. User opens context menu not in the selected table.
-			if ( evt.name === 'mouseup' && !isOutsideTable( evt.data.getTarget() ) && !isSameTable( selectedTable, table ) ) {
-				return true;
-			}
+			// 2. User opens context menu outside of selection.
+			// Use 'mousedown' for Firefox, as it doesn't fire 'mouseup' when mouse is released in context menu.
+			return isProperEvent( evt.name ) && !isOutsideTable( evt.data.getTarget() ) &&
+				!isInSelectedCell( evt.data.getTarget(), fakeSelectedClass, table );
+		}
 
-			return false;
+		function isProperEvent( evtName ) {
```

Comment:
`isProperMouseEvent` IMO sounds better. Also if this function is not used in any other place, I'd move that function into `canClearSelection` or even inline it as a variable.